### ISSUE TITLE
Linters and vim mode(s) don't get along

### ISSUE THREAD

**Issue by onyxraven** [rubyide/vscode-ruby#252]
### Your environment

- `vscode-ruby` version: 0.16.0 (and 0.15.0 it turns out)
- Ruby version: 2.4.3
- VS Code version: Version 1.19.3 (1.19.3)
- Operating System: OSX 10.13.3 (17D47)
- Hardware (optional): MacBook Pro (Retina, 15-inch, Mid 2014)

Make sure you have `ruby`, `ruby-debug-ide` and `ruby-debug-basex19` installed before submitting your issue -- thank you !

### Expected behavior

When using Vim or amVim extensions along with Ruby linters enabled, typing should not be delayed every character for lints to complete.

### Actual behavior

It seems like for every linter enabled, they are running on the main input thread and blocking the inputs.  This looks to even get in between the keypress and the autocomplete/suggestions.

### Steps to reproduce the problem

It seems like using either of those extensions, along with any enabled linters (which arguably do hook all editor inputs for modes) cause the issue to show very obviously.  Editing other file types with the extensions is not a problem.  Disabling all linters makes inputs work again without delay.  

----------

I'm happy to help with more investigation, but I just need some guidance on where to look next.

**Comment by onyxraven** [comment]
I just downgraded to 0.15 (using the helpful link [in a different issue](https://github.com/rubyide/vscode-ruby/issues/251#issuecomment-360806583)) and it seems to be still happening in the same way. So I'm thinking it has something to do with the 1.19.3 update.  I'll check the issue board for the main VSCode project to see if there is anything related.

**Comment by castwide** [comment]
I don't use Vim or amVim, but I had a similar problem. I got the best performance (short of no linters at all) using RuboCop alone with the latest gem.

**Comment by MythicalFish** [comment]
Same issue here, can confirm that disabling the linters solves the lag.

**Comment by wingrunr21** [comment]
Can some of you post your linter configs?

I also don't see this issue using VIM + Rubocop. I'm wondering if the changes made in #232 might be affecting something there (although linter performance wasn't great before that either).

**Comment by cyangOL** [comment]
I have the same issue. Typing is noticeably slow when vim and ruby are the only enabled plugin.

**Comment by onyxraven** [comment]
in 1.6.0 I was able to reproduce with the following config

```javascript
    "ruby.lint": {
        "rubocop": false,
        "ruby": true, //Runs ruby -wc
        "fasterer": false,
        "debride": false,
        "ruby-lint": false
    },
```

`ruby -wc` is fastish but it still would delay typing input.

I downgraded to 1.5 and the delay has seemed to go away this time.  I'm still using the Vim plugin.  I'll see if I can gather any other situational info

**Comment by MythicalFish** [comment]
Any updates on this? Would be nice to reenable linters. :)

**Comment by onyxraven** [comment]
Master is fixed with the changes in #264
